src/main/java/ModelsTest/UserTest/UserDaoTest.java

```diff
@@ -0,0 +1,93 @@
+package ModelsTest.UserTest;
+
+import Models.User.DuplicateEmailPresentException;
+import Models.User.User;
+import Models.User.UserDao;
+import Models.User.UserNotFoundException;
+import jdk.nashorn.internal.ir.annotations.Ignore;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class UserDaoTest {
+
+    private final ByteArrayOutputStream outputStreamCaptor = new ByteArrayOutputStream();
+
+    @BeforeEach
+    void setUp() {
+        System.setOut(new PrintStream(outputStreamCaptor));
+    }
+
+    @AfterEach
+    void tearDown() throws IOException {
+        outputStreamCaptor.close();
+        // TODO: Revert changes made to database in unit test 'Given_ValidUser_Then_InsertUser'
+    }
+
+    @Ignore
+    @Test
+    void Given_ValidUser_Then_InsertUser() throws DuplicateEmailPresentException {
+        // Given
+        User user = new User(12, "Charlie", "Williams,", "charlie@uwe.ac.uk", new byte[16], new byte[16]);
+
+        // When
+        UserDao.insertUser(user);
+
+        // Then
+        assertEquals("User inserted", outputStreamCaptor.toString());
+    }
+
+    @Ignore
+    @Test
+    void Given_DuplicateUser_Then_ThrowException() {
+        // Given
+        User user = new User(12, "Charlie", "Williams,", "charlie@uwe.ac.uk", new byte[16], new byte[16]);
+
+
+        // When
```

Comment:
If we tear down the DB between tests and rebuild it anew, this test will currently not throw a duplicate email error if I'm not mistaken - should the user be inserted in 'When'?

Reply:
I imagine this unit test will need some work. So in this test, we'll need to insert a user successfully, and then try to insert the same user again.

So I think we would have it in both `Given `and `When`.

I'll look for some other syntax for `assertThrows` because that has the `UserDao.insertUser(user)` which should be in `When`.